### ISSUE TITLE
`react-dom` version conflict with RN 0.45.1

### ISSUE THREAD

**Issue by tmeasday** [storybooks/storybook#1284]
The CLI's [hardcoded version](https://github.com/storybooks/storybook/blob/master/lib/cli/generators/REACT_NATIVE/index.js#L27) of `react-dom` is incompatible with the version of React installed by RN 0.45.1.

Reproduction

```
# install latest RN
npm install -g react-native-cli

# create RN app
react-native init MyApp

# add storybook
cd MyApp
getstorybook

# try running storybook
npm run storybook
```

You will get an error due to incompability between `react-dom@15.5.4` and `react@16.0.0-alpha.12`:

```
ERROR in ./~/react-dom/lib/ReactMount.js
Module not found: Error: Can't resolve 'react/lib/React' in '/private/tmp/TesterMeTester/node_modules/react-dom/lib'
 @ ./~/react-dom/lib/ReactMount.js 17:12-38
 @ ./~/react-dom/lib/ReactDOM.js
 @ ./~/react-dom/index.js
 @ ./~/@storybook/ui/dist/modules/ui/routes.js
 @ ./~/@storybook/ui/dist/modules/ui/index.js
 @ ./~/@storybook/ui/dist/index.js
 @ ./~/@storybook/react-native/dist/manager/index.js
 @ multi ./storybook/addons.js ./~/@storybook/react-native/dist/manager/index.js

...etc..
```

Update the reference of `react-dom` in `package.json` and it should run again (modulo https://github.com/storybooks/storybook/pull/1280)

**Comment by shilman** [comment]
@tmeasday I have confirmed the problem and the fix. 👍

**Comment by tmeasday** [comment]
OK, I guess we should figure out the version of `react-dom` based on the `react` version in `package.json`?

**Comment by Fanghao** [comment]
Is it fixed? I still get the error when running 3.1.4.

**Comment by tmeasday** [comment]
Hi @Fanghao, no, but it is easily fixed on a case-by-case basis. Just run `npm upgrade react-dom` in your app.

**Comment by tmeasday** [comment]
Fixed in 3.1.6!
